### ISSUE TITLE
New feature: go to column

### ISSUE THREAD

**Issue by ComputingFroggy** [geany/geany#869]
Hi,

I will be really nice to have the feature "go to column" to be able to easily go to a specific column (say 1500) for text files with long lines (usually data exchange files).

**Comment by elextr** [comment]
The problem is, whats a "column"?  For ascii characters its a byte number, for Unicode is it a code point count where code points can use several bytes or you can still mean a byte number, or maybe you mean a visual column which with combining characters can be several code points which are several bytes which ....

**Comment by ComputingFroggy** [comment]
A column is what is counted in the "col" counter in the status bar at the bottom of Geany editor ! ;-)

Also there's a strange thing lines are counted starting at 1 when column are counted starting at 0 !
It would be more interesting to start counting columns at 1 (or at least to get an option to chose to count columns from 1 or from 0).

Cheers,
L@u

**Comment by codebrainz** [comment]
@ComputingFroggy see http://sourceforge.net/p/geany/bugs/505/

**Comment by elextr** [comment]
See [the fine manual](http://www.geany.org/manual/current/index.html#statusbar-templates ) for how to change the display of "column".

A "column" is what the editing component we use says it is, which is not all that clear, but seems to most closely approximate code points, except tabs are counted as their display width, and its not clear how combining characters are handled.  All in all its a mess, but its unlikely to change.

**Comment by ComputingFroggy** [comment]
Anyhow, I am not debating about the column count here, I just would like to have in my favourite editor a function to go to a specific column ... as it is possible to perform on most modern (and even some ancient) editors !

**Comment by elextr** [comment]
Well, any PR to add this would have to make it very clear in the manual what it actually does.

What the manual is saying is that the default template contains a tab character at those positions (otherwise they don't show in the manual).  It is not saying that \t is translated into a tab.  You can just put an actual tab character in the template.

**Comment by ComputingFroggy** [comment]
My mistake, I falsely understood there was a translation of \t into TAB (like it is often the case in programming languages). 
Now, I understand my misunderstanding. ;-)

**Comment by jesus2099** [comment]
For instance, in EmEditor, the command Jump to line… was renamed long time ago to Jump… (Ctrl+G).
The column fields defaults to 1, to mimic the Jump to line behaviour:

![EmEditor: Jump…](https://user-images.githubusercontent.com/1401086/105148584-d7972080-5b02-11eb-9063-689e4a8aedab.png)

So, if you want to jump to a specific line:

1. Press Ctrl+G
2. Type line number
3. Press Enter

And, if you want to jump to specific line and column:

1. Press Ctrl+G
2. Type line number
3. Press Tab
4. Type column number
5. Press Enter

Note that Geany status bar column numbers should be fixed first (#2733).

**Comment by elextr** [comment]
Yes adding it to the goto line dialog like that would be possible, just needs somebody to do it.
